core/src/main/java/org/openedx/core/feature/FeatureRequest.kt

```diff
@@ -0,0 +1,33 @@
+package org.openedx.core.feature
+
+/**
+ * Base sealed class representing a request for a feature decision.
+ */
+sealed class FeatureRequest<T> {
+    abstract val key: String
+}
+
+/**
+ * Request for a Boolean feature flag.
+ */
+data class FeatureFlagRequest(
+    override val key: String,
+) : FeatureRequest<Boolean>()
+
+/**
+ * Request for an A/B test decision.
+ */
+data class AbTestRequest(
+    override val key: String,
+) : FeatureRequest<String>()
+
+/**
+ * Request for a remote config value.
+ *
+ * @param key The primary key representing the remote config feature.
+ * @param flagKey The secondary key representing the flag variant.
+ */
+data class RemoteConfigRequest(
+    override val key: String,
+    val flagKey: String,
+) : FeatureRequest<String>()
```

Comment:
IMO, it's better to combine all the requests into a single call, return the complete dataset, and provide the flexibility to traverse the data at the call site.